app/main.py

```diff
@@ -1,25 +1,110 @@
 from abc import ABC
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int,
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: Any,
```

Comment:
Owner shouldn't be Any, it's too wide. Add more descriptive typehint, please :)
Fix it everywhere.

Reply:
Done